libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -1007,6 +1011,16 @@ void CompilerUtils::cleanHigherOrderBits(IntegerType const& _typeOnStack)
 		m_context << ((u256(1) << _typeOnStack.numBits()) - 1) << Instruction::AND;
 }
 
+void CompilerUtils::leftShiftNumberOnStack(int _shiftFactor)
+{
+	m_context << (u256(1) << _shiftFactor) << Instruction::MUL;
+}
+
+void CompilerUtils::rightShiftNumberOnStack(int _shiftFactor, bool _isSigned)
+{
+	m_context << (u256(1) << _shiftFactor) << Instruction::SWAP1 << (_isSigned ? Instruction::SDIV : Instruction::DIV);
```

Comment:
Please add an assert that `0 <= shiftFactor < 256` to avoid the boost multiprecision shift bug.